pkg/providers/snow/reconciler/clientbuilder_test.go

```diff
@@ -0,0 +1,99 @@
+package reconciler_test
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/yaml"
+
+	"github.com/aws/eks-anywhere/pkg/providers/snow/reconciler"
+)
+
+func TestBuildSnowAwsClientMap(t *testing.T) {
+	tests := []struct {
+		name       string
+		secretFile string
+		wantErr    string
+	}{
+		{
+			name:       "valid",
+			secretFile: "testdata/valid_secret",
+			wantErr:    "",
+		},
+		{
+			name:       "invalid ini format",
+			secretFile: "testdata/invalid_secret_ini_format",
+			wantErr:    "loading values from credentials: key-value delimiter not found: 1.2.3.5",
+		},
+		{
+			name:       "missing aws access key",
+			secretFile: "testdata/invalid_secret_missing_aws_access_key",
+			wantErr:    "parsing configuration for 1.2.3.5: unable to set aws_access_key_id",
+		},
+		{
+			name:       "missing aws secret key",
+			secretFile: "testdata/invalid_secret_missing_aws_secret_key",
+			wantErr:    "parsing configuration for 1.2.3.5: unable to set aws_secret_access_key",
+		},
+		{
+			name:       "missing region",
+			secretFile: "testdata/invalid_secret_missing_region",
+			wantErr:    "parsing configuration for 1.2.3.5: unable to set region",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+			ctx := context.Background()
+
+			secret, err := readSecret(tt.secretFile)
```

Comment:
I feel like it would be more clear to just define the secrets with structs in the test code and avoid testdata files